backend/server/services/UserInfoService.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file Defines the UserInfoService
+ */
+
+const admin = require('firebase-admin');
+const winston = require('winston');
+
+/**
+ * Creates a new canvas based on a UserInfoService request
+ * @param {ExpressRequest} request An express request object representing the request
+ * @param {ExpressResponse} response An express response object representing the response
+ * @returns {void}
+ */
+const handleRequest = (request, response) => {
+  winston.info(
+    'UserInfoService.handleRequest - handling a new request: %j',
+    request.body
+  );
+
+  if(typeof request.body.uid !== "string") {
+    winston.error('UserInfoService.handleRequest - invalid uid param, must be a String');
+    response.status(500)
+      .send('Invalid uid param.');
+    return;
+  }
+
+  try {
+    admin.auth().getUser(request.body.uid)
+      .then((userRecord) => {
+        let message = "";
+        admin.database().ref('/users/' + request.body.uid)
+          .transaction((oldValue) => {
+            if(oldValue)
+            {
+              winston.info(
+                'UserInfoService.handleRequest - user info for user %s already saved',
+                request.body.uid
+              );
+
+              message = "user info already existed, overwriting"
+            } else {
+              winston.info(
+                'UserInfoService.handleRequest - user info for user %s not saved, overwriting',
+                request.body.uid
+              );
+
+              message = "new user info saved"
+            }
+
+            winston.info(
+              'UserInfoService.handleRequest - saving user info for user %s',
+              request.body.uid
+            );
+
+            let providerData = [];
+            userRecord.providerData.forEach((singleProvData) => {
+              providerData.push({
+                displayName: singleProvData.displayName ? singleProvData.displayName : null,
+                email: singleProvData.email ? singleProvData.email : null,
+                photoURL: singleProvData.photoURL ? singleProvData.photoURL : null,
+                providerId: singleProvData.providerId,
```

Comment:
Will displayName or email ever be null? Or is this ternary operator working as redundancy?  If so, should a similar operation should be done on the providerId (singleProvData.providerId ? singleProvData.providerId : null)?